Gems/ROS2/Code/Include/ROS2/Sensor/Events/TickBasedSource.h

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/Component/TickBus.h>
+#include <AzCore/EBus/Event.h>
+#include <ROS2/Sensor/Events/SensorEventSource.h>
+
+namespace ROS2
+{
+    //! Class implementing system TickBus (draw calls) as sensor event source. Source event (ROS2::SensorEventSource) is signalled based on
+    //! system ticks. This event source can de enabled / disabled using m_publishingEnabled member of SensorConfiguration
```

Comment:
There are cases where the source could be active even though it is not publishing, such as simply acquiring data to visualize it or process simulation-side only.